website/static/js/keen.js

```diff
@@ -0,0 +1,143 @@
+                'returning': returning,
+                'pageUrl': document.URL,
+                'keen': {
+                    'addons': [
+                        {
+                            'name': 'keen:ip_to_geo',
+                            'input': {
+                                'ip': 'ipAddress'
+                            },
+                            'output': 'ipGeoInfo'
+                        },
+                        {
+                            'name': 'keen:ua_parser',
+                            'input': {
+                                'ua_string': 'userAgent'
+                            },
+                            'output': 'parsedUserAgent'
+                        },
+                        {
+                            'name': 'keen:referrer_parser',
+                            'input': {
+                                'referrer_url': 'referrer.url',
+                                'page_url': 'pageUrl'
+                            },
+                            'output': 'referrer.info'
+                        }
+                    ]
+                }
+                // 'daysSinceFirstVisit':'',
+                // 'daysSinceLastVisit': '',
+                //'resolution': '' //temp user agent stuff
+            };
+
+            if(ctx.currentUser){
+                visit.currentUser = ctx.currentUser;
+            }
+
+            this.keenClient.addEvent('visits', visit, function(err){
+                if(err){
+                    throw 'Error sending Keen data: ' + err;
+                }
+            });
+        }
+    },
+
+    trackPageView: function(){
+        this.createOrUpdateKeenSession();
+        var ctx = window.contextVars;
+        var pageView = {
+            'pageUrl': document.URL,
+            'keen': {
+                'addons': [
+                    {
+                        'name' : 'keen:url_parser',
+                        'input' : {
+                            'url' : 'pageUrl'
+                        },
+                        'output' : 'parsedPageUrl'
+                    }
+                ]
+            },
+            'keenId': this.getOrCreateKeenId(),
+            'sessionId': $.cookie('keenSessionId'),
+            'pageTitle': document.title
+            //'generation_time': ''
+            //'timeSpend': ''
+        };
+        if(ctx.node){
+            pageView.node = {
+                'id': ctx.node.id,
+                'title': ctx.node.title,
+                'type': ctx.node.category,
+                'tags': ctx.node.tags
+            };
+        }
+        if(ctx.currentUser){
+            pageView.user = ctx.currentUser;
+        }
+
+        this.keenClient.addEvent('pageviews', pageView, function(err){
+            if(err){
+                throw 'Error sending Keen data: ' + err;
+            }
+        });
+    },
+
+    trackCustomEvent: function(eventCollection, eventData){
```

Comment:
This was intended for future extensibility by other devs who need more specific things tracked. I can take it out for now if needed.